bootstrap/src/main/java/io/airlift/bootstrap/Bootstrap.java

```diff
@@ -209,9 +210,15 @@ public Injector initialize()
 
         // initialize logging
         if (logging != null) {
-            log.info("Initializing logging");
-            LoggingConfiguration configuration = configurationFactory.build(LoggingConfiguration.class);
-            logging.configure(configuration);
+            if (loggingConfigured) {
+                log.warn("Logging already configured; ignoring new configuration.");
```

Comment:
I don't think we want to log a warning when the configuration is the same. It would be better to silently ignore it. Otherwise, this is effectively making it a mistake to initialize logging twice, with the result being that we should change any such application code to not initialize the second time. But if we were to do that, we wouldn't need this change at all (except to catch mistakes).

Reply:
I would then change to info, then.

Comparing configurations is something I'm trying to avoid, as it's not trivial, and doesn't bring much value. We're just trying to avoid adding the appender twice.